xmake/rules/python/cython/xmake.lua

```diff
@@ -0,0 +1,60 @@
+--!A cross-platform build utility based on Lua
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+-- Copyright (C) 2015-present, TBOOX Open Source Group.
+--
+-- @author      Wu, Zhenyu
+-- @file        xmake.lua
+--
+
+rule("python.cython")
+    set_extensions(".py", ".pyx")
+    before_buildcmd_file(function(target, batchcmds, sourcefile, opt)
+        local language = target:extraconf("rules", "python.cython", "language")
+        local ext
+        local arg = "-3"
+        if language == "c" then
+            ext = "c"
+            local rule = target:rule("python.cython"):clone()
+            rule:add("deps", "c", {order = true})
+            target:rule_add(rule)
```

Comment:
we should set `target:add("deps", "c")` in on_load() instead of before_buildcmd_file.

and we need not use rule_add